content/javascript/concepts/optional-chaining/optional-chaining.md

```diff
@@ -0,0 +1,167 @@
+```
+
+## Example
+This person object lists an individual's details and containing properties, nested objects, an array, and a method. Optional chaining for each conditional evaluation is listed below. 
+
+```js
+const person = {
+  name: "Alice",
+  gender: "Female",
+  age: 12,
+  address: {
+    street: "1111 Palm Ave", 
+    city: "Broken Arrow",
+    state: "Oklahoma"
+  },
+  favoriteFoods: ["pizza", "ice cream", "cake"], 
+  commonPhrase: function() {
+    return `${this.name} always says "I love ${this.address.state}."`;
+  }
+};
+```
+
+## Accessing Object Properties
+To search for the `state` object of `person`:
+
+```js
+//Regular syntax
+const state = person && person.address && person.address.state;
+console.log(`State is: ${state}`);
+
+//This can be rewritten as:
+const state = person?.address?.state;
+console.log(`State is: ${state}`);
+
+//The output for both is: 
+State is: Oklahoma
+```
+
+## Accessing Arrary Elements
+To search for the `cake` in the `favoriteFoods` array of `person`:
+
+```js
+//Regular Syntax
+const food = person && person.favoriteFoods && person.favoriteFoods.find(item => item === "cake");
+console.log(`Favorite Food is: ${food}`);
+
+//This can be rewritten as:
+const food = person?.favoriteFoods.find(item => item === "cake");
+console.log(`Favorite Food is: ${food}`)
+
+//The output for both is: 
+Favorite Food is: cake
+```
+
+## Accessing Object Functions
+To determine if the `commonPhrase` function exists in `person` before invoking it:
+
+```js
+//Regular Syntax
+if (person && typeof person.commonPhrase === "function") {
+  const phrase = person.commonPhrase();
+console.log(`${phrase}`);
+}
+
+//This can be rewritten as:
+const phrase = person?.commonPhrase?.();
+console.log(`${phrase}`);
+
+//The output for both is: 
+Alice always says "I love Oklahoma."
+```
+
+
+## Codebyte Example
+
+Run the following example and compare the regular syntax of conditional statements and optional chaining on a more complex data set.
+
+```codebyte/js
+const people = [
+  {
+    name: "Blake",
+    gender: "Male",
+    age: null,
+    address: {
+      street: "456 Maple Dr",
+      city: "Austin",
+      state: "Texas"
+    }
+  },
+  {
+    name: "Eve",
+    gender: "Female",
+    age: 15,
+    address: {
+      street: "789 Birch Ln",
+      city: "",
+      state: "Colorado"
+    }
+  },
+  null,
+  {
+    name: "David",
+    gender: "Male",
+    age: 72,
+    address: {
+      street: "123 Acorn St",
+      city: "Tampa",
+      state: "Florida"
+    }
+  },
+  {
+    name: "Jospeh",
+    gender: "Male",
+    age: 9,
+    address: {
+      street: "987 Pine Ave",
+      city: "Taos",
+      state: "New Mexico"
+    }
+  }
+];
+
+people.forEach((person, index) => {
+  const city = person && person.address && person.address.city;
+  console.log(`Person ${index + 1}: City: ${city}`);
+});
+
+people.forEach((person, index) => {
+const age = person?.age;
+console.log(`Person ${index + 1}: Age: ${age}`);
+});
```

Comment:
It would be better to directly specify that this is the "regular syntax" for conditional checks, and then show how optional chaining simplifies this:

```suggestion
// Regular syntax for checking if a person's city exists
people.forEach((person, index) => {
  const city = person && person.address && person.address.city;
  console.log(`Person ${index + 1}: City: ${city}`);
});

// Optional chaining for simpler access
people.forEach((person, index) => {
const age = person?.age;
console.log(`Person ${index + 1}: Age: ${age}`);
});
```

Reply:
The output of the above code is as follows - 

```
Person 1: City: Austin
Person 2: City: 
Person 3: City: null
Person 4: City: Tampa
Person 5: City: Taos
Person 1: Age: null
Person 2: Age: 15
Person 3: Age: undefined
Person 4: Age: 72
Person 5: Age: 9
```

Ideally, the output of both syntaxes should be the same so that the learner understands that they can be used alternatively. Update the code to reflect the same.